shell/scripts/test-plugins-build.sh

```diff
@@ -2,22 +2,132 @@
 
 set -e
 
-echo "Checking plugin build"
+CYAN="\033[96m"
+YELLOW="\033[93m"
+RESET="\033[0m"
+BOLD="\033[1m"
+NORMAL="\033[22m"
+
+echo -e "${BOLD}${CYAN}Extensions framework build tests${RESET}"
 
 SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
 BASE_DIR="$( cd $SCRIPT_DIR && cd ../.. & pwd)"
 SHELL_DIR=$BASE_DIR/shell/
 
-echo ${SCRIPT_DIR}
+# ---- Options ----------------------------------------------------------------------------------------------------
+
+KEEP_FOLDER=false
+KEEP_PKG=false
+SKIP_IN_TREE=false
+SKIP_SETUP=false
+SKIP_STANDALONE=false
+
+REG_USER=admin
+REG_PWD=admin
+
+usage() {
+  echo "Usage: $0 [<options>] [plugins]"
+  echo " options:"
+  echo "  -k           Keep the folder for the test application after run"
+  echo "  -e           Keep the test extension after run (in-tree tests)"
+  echo "  -i           Skip in-tree tests"
+  echo "  -s           Skip standalone tests"
+  echo "  -x           Skip test setup"
+  echo "  -u <name>    Verdaccio user name for publishing (default: admin)"
+  echo "  -p <name>    Verdaccio password for publishing"
+  exit 1
+}
+
+while getopts "hkeisr:p:" opt; do
+  case $opt in
+    h)
+      usage
+      ;;
+    k)
+      KEEP_FOLDER=true
+      echo "Folder for test appliaction will not be deleted after test run"
+      ;;
+    e)
+      KEEP_PKG=true
+      echo "Folder for test extension will not be deleted after test run"
+      ;;
+    i)
+      SKIP_IN_TREE=true
+      echo "In-tree tests will not be run"
+      ;;
+    s)
+      SKIP_STANDALONE=true
+      echo "Standalone tests will not be run"
+      ;;
+    x)
+      SKIP_SETUP=true
+      echo "Setup will be skipped"
+      ;;
+    r)
+      REG_USER=${OPTARG}
+      echo "Registry user: ${REG_USER}"
+      ;;
+    p)
+      REG_PWD=${OPTARG}
+      ;;
+    *)
+      usage
+      ;;
+  esac
+done
+
+shift $((OPTIND-1))
+
+# ---- Cleanup ----------------------------------------------------------------------------------------------------
+# Cleanup called on exit and if process if stopped via CTRL+C
+cleanup() {
+  echo ""
 
-SKIP_SETUP="false"
-SKIP_STANDALONE="false"
+  set +e
 
-if [ "$1" == "-s" ]; then
-  SKIP_SETUP="true"
-fi
+  rm -f ~/.npmrc.bak
 
-if [ $SKIP_SETUP == "false" ]; then
+  if [ -n "${TEMP_APP_FOLDER}" ]; then
+    if [ ${KEEP_FOLDER} == false ]; then
+      echo "Removing test application folder"
+      rm -rf ${TEMP_APP_FOLDER}
+    else
+      echo -e "${BOLD}${CYAN}Test application was not deleted - folder:"
+      echo -e "${TEMP_APP_FOLDER}${RESET}"
+    fi
+  fi
+
+  if [ ${KEEP_PKG} == false ]; then
+    rm -rf ./pkg/test-pkg
+  fi
+
+  # Restore shell version
+  if [ -n "${SHELL_VERSION}" ]; then
+    sed -i.bak -e "s/\"version\": \"7.7.7\",/\"version\": \"${SHELL_VERSION}\",/g" ${SHELL_DIR}/package.json
```

Comment:
In case of multiple publications the version value should be increased or Verdaccio throws an error. Although we could clean it up and start again, that's quite time consuming.

Reply:
We could alternatively add a command after the login where we remove the libraries, as it's quite fast and does not return error if already removed.

The libraries cannot be added in list, unfortunately.

```
npm unpublish --force @rancher/shell --registry=http://localhost:4873/ 
```

Reply:
It should already clean Verdaccio of old packages - its does this directly by deleting files from Verdaccio's local storage.

For npm unpublish, would need to check package is present and version number

Reply:
If there's no need to add any flag of any type, it seems that the packages are not removed. I had this issue also with the previous version of the script, same message of above.

Reply:
Updated to use npm unpublish now